server.js

```diff
@@ -2,25 +2,161 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import Thought from './models/thought'
+import Comment from './models/comment'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, {
+  useNewUrlParser: true, useUnifiedTopology: true,
+  useFindAndModify: false
+})
 mongoose.Promise = Promise
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// overridden when starting the server. For example: PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const listEndpoints = require('express-list-endpoints')
+
+// Add middlewares to enable cors and json body parsing,
+// and handling if API service is unavailable
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+// Error messages
+const ERR_NO_THOUGHTS = 'There are no happy thoughts yet'
+const ERR_NO_PAGE = 'Requested page not found, could not get thoughts'
+const ERR_GET_THOUGHTS = 'Invalid request, could not get thoughts'
+const ERR_POST_THOUGHT = 'Invalid request, could not save thought'
+const ERR_POST_LIKE = 'Invalid request, could not save like'
+const ERR_GET_COMMENTS = 'Invalid request, could not get comments'
+const ERR_POST_COMMENT = 'Invalid request, could not save comment'
 
-// Start defining your routes here
+// Root endpoint
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Endpoint returning 20 thoughts
+app.get('/thoughts', async (req, res) => {
```

Comment:
I really like this `express-list-endpoints` but for this project the documentation says that "/" should return the 20 thoughts

Reply:
I see, but would you say that is restful?

Reply:
> I really like what you've done with this project! Just take another look at the documentation and change your endpoints' names accordingly :) Apart from that - keep up the good work! ⭐

Approving both `/` and `/thoughts`, sorry for the confusion!

Reply:
Yay, thank you 🎉